### ISSUE TITLE
strconv.Atoi: parsing [...] invalid syntax

### ISSUE THREAD

**Issue by vertexvaar** [dfeyer/flow-debugproxy#18]
In my code i have an assoc array with 78 entries (for writing data to a CSV file) and every time i try to step over initializing the array the flow-debugproxy quits with the following (vendor and package name edited) error Message

```
strconv.Atoi: parsing "del\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"78\" fullname=\"$transactions[78]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"79\" fullname=\"$transactions[79]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"80\" fullname=\"$transactions[80]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"81\" fullname=\"$transactions[81]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"82\" fullname=\"$transactions[82]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"83\" fullname=\"$transactions[83]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"84\" fullname=\"$transactions[84]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"85\" fullname=\"$transactions[85]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"86\" fullname=\"$transactions[86]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"87\" fullname=\"$transactions[87]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"88\" fullname=\"$transactions[88]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"89\" fullname=\"$transactions[89]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"90\" fullname=\"$transactions[90]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"91\" fullname=\"$transactions[91]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"92\" fullname=\"$transactions[92]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"93\" fullname=\"$transactions[93]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"94\" fullname=\"$transactions[94]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"95\" fullname=\"$transactions[95]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"96\" fullname=\"$transactions[96]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"97\" fullname=\"$transactions[97]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"98\" fullname=\"$transactions[98]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"99\" fullname=\"$transactions[99]\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property></property><property name=\"$zip\" fullname=\"$zip\" type=\"uninitialized\"></property><property name=\"$this\" fullname=\"$this\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Command\\ReportCommandController\" children=\"1\" numchildren=\"10\" page=\"0\" pagesize=\"100\"><property name=\"config\" fullname=\"$this-&gt;config\" facet=\"protected\" type=\"array\" children=\"1\" numchildren=\"7\"></pstrconv.Atoi: parsing "del\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"78\" fullname=\"$transactions[78]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"79\" fullname=\"$transactions[79]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"80\" fullname=\"$transactions[80]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"81\" fullname=\"$transactions[81]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"82\" fullname=\"$transactions[82]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"83\" fullname=\"$transactions[83]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"84\" fullname=\"$transactions[84]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"85\" fullname=\"$transactions[85]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"86\" fullname=\"$transactions[86]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"87\" fullname=\"$transactions[87]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"88\" fullname=\"$transactions[88]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"89\" fullname=\"$transactions[89]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"90\" fullname=\"$transactions[90]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"91\" fullname=\"$transactions[91]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"92\" fullname=\"$transactions[92]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"93\" fullname=\"$transactions[93]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"94\" fullname=\"$transactions[94]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"95\" fullname=\"$transactions[95]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"96\" fullname=\"$transactions[96]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"97\" fullname=\"$transactions[97]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"98\" fullname=\"$transactions[98]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property><property name=\"99\" fullname=\"$transactions[99]\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Model\\Transaction\" children=\"1\" numchildren=\"13\"></property></property><property name=\"$zip\" fullname=\"$zip\" type=\"uninitialized\"></property><property name=\"$this\" fullname=\"$this\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Command\\ReportCommandController\" children=\"1\" numchildren=\"10\" page=\"0\" pagesize=\"100\"><property name=\"config\" fullname=\"$this-&gt;config\" facet=\"protected\" type=\"array\" children=\"1\" numchildren=\"7\"></property><property name=\"marketplaceFactory\" fullname=\"$this-&gt;marketplaceFactory\" facet=\"protected\" type=\"object\" classname=\"Onedrop\\CyzouMarketplace\\Domain\\Factory\\MarketplaceFactory\" children=\"1\" numchildren=\"3\"></property><property name=\"request\" fullname=\"$this-&gt;request\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\Request\" children=\"1\" numchildren=\"7\"></property><property name=\"response\" fullname=\"$this-&gt;response\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\Response\" children=\"1\" numchildren=\"4\"></property><property name=\"arguments\" fullname=\"$this-&gt;arguments\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Mvc\\Controller\\Arguments\" children=\"1\" numchildren=\"2\"></property><property name=\"commandMethodName\" fullname=\"$this-&gt;commandMethodName\" facet=\"protected\" type=\"string\" size=\"15\" encoding=\"base64\"><![CDATA[Z2VuZXJhdGVDb21tYW5k]]></property><property name=\"objectManager\" fullname=\"$this-&gt;objectManager\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\ObjectManagement\\ObjectManager\" children=\"1\" numchildren=\"9\"></property><property name=\"commandManager\" fullname=\"$this-&gt;commandManager\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\CommandManager\" children=\"1\" numchildren=\"5\"></property><property name=\"output\" fullname=\"$this-&gt;output\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\ConsoleOutput\" children=\"1\" numchildren=\"5\"></property><property name=\"Flow_Injected_Properties\" fullname=\"$this-&gt;Flow_Injected_Properties\" facet=\"public\" type=\"array\" children=\"1\" numchildren=\"4\"></property></property></response>": invalid syntaxroperty><property name=\"marketplaceFactory\" fullname=\"$this-&gt;marketplaceFactory\" facet=\"protected\" type=\"object\" classname=\"Company\\HugeNeosFlowAppl\\Domain\\Factory\\MarketplaceFactory\" children=\"1\" numchildren=\"3\"></property><property name=\"request\" fullname=\"$this-&gt;request\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\Request\" children=\"1\" numchildren=\"7\"></property><property name=\"response\" fullname=\"$this-&gt;response\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\Response\" children=\"1\" numchildren=\"4\"></property><property name=\"arguments\" fullname=\"$this-&gt;arguments\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Mvc\\Controller\\Arguments\" children=\"1\" numchildren=\"2\"></property><property name=\"commandMethodName\" fullname=\"$this-&gt;commandMethodName\" facet=\"protected\" type=\"string\" size=\"15\" encoding=\"base64\"><![CDATA[Z2VuZXJhdGVDb21tYW5k]]></property><property name=\"objectManager\" fullname=\"$this-&gt;objectManager\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\ObjectManagement\\ObjectManager\" children=\"1\" numchildren=\"9\"></property><property name=\"commandManager\" fullname=\"$this-&gt;commandManager\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\CommandManager\" children=\"1\" numchildren=\"5\"></property><property name=\"output\" fullname=\"$this-&gt;output\" facet=\"protected\" type=\"object\" classname=\"Neos\\Flow\\Cli\\ConsoleOutput\" children=\"1\" numchildren=\"5\"></property><property name=\"Flow_Injected_Properties\" fullname=\"$this-&gt;Flow_Injected_Properties\" facet=\"public\" type=\"array\" children=\"1\" numchildren=\"4\"></property></property></response>": invalid syntax
```

**Comment by fteuber** [comment]
*Could* be the same problem I encounter quite often when the XDebug-Response (XML) is too large for one chunk. It will then be sent in multiple chunks but the proxy expects only one chunk with a leading integer (the size) and a valid/wellformed XML. The 2nd chunk is an invalid XML with no leading integer causing the problem.

**Comment by dfeyer** [comment]
Thanks for reporting, it's a known issue, but I never found the time to update the parser to support this multi chunk parsing. I will try to work on it after my holiday, hope to have a fix for next month.

**Comment by jdoubleu** [comment]
I'm having the same issue.
Although it works sometimes. As soon as I set the `xdebug.remote_log` config it works, too.
Maybe it has something todo with internal buffering in xdebug.

The log below shows the output of the debugproxy in *very verbose* (`-vv`) mode.
As you can see the requested context (`context_get -i 24 -d 0 -c 1`) could not be decoded.
Probably because of the multiple chunks, @fteuber mentioned.
The [`Proxy.pipe` function](https://github.com/dfeyer/flow-debugproxy/blob/master/xdebugproxy/xdebugproxy.go#L85) does only work with one packet and [writes it into a buffer](https://github.com/dfeyer/flow-debugproxy/blob/master/xdebugproxy/xdebugproxy.go#L99).

```
...
Debugger >>> IDE
================
Raw protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="run" transaction_id="20" status="break" reason="ok">
  <xdebug:message filename="file:///var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/GoCardTeam_GoCardApi_Http_v1_CORSInterceptionComponent.php" lineno="25"/>
</response>

Umpa Lumpa can help you, he know the mapping
>>> /var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/GoCardTeam_GoCardApi_Http_v1_CORSInterceptionComponent.php
>>> /var/www/html/Packages/Application/GoCardTeam.GoCardApi/Classes/Http/v1/CORSInterceptionComponent.php

Processed protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="run" transaction_id="20" status="break" reason="ok">
  <xdebug:message filename="file:///var/www/html/Packages/Application/GoCardTeam.GoCardApi/Classes/Http/v1/CORSInterceptionComponent.php" lineno="25"/>
</response>


IDE >>> Debugger
================
Raw protocol:
stack_get -i 21

Processed protocol:
stack_get -i 21


Debugger >>> IDE
================
Raw protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="stack_get" transaction_id="21">
  <stack where="GoCardTeam\GoCardApi\Http\v1\CORSInterceptionComponent->handle" level="0" type="file" filename="file:///var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/GoCardTeam_GoCardApi_Http_v1_CORSInterceptionComponent.php" lineno="25"/>
  <stack where="Neos\Flow\Http\Component\ComponentChain->handle" level="1" type="file" filename="file:///var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Component_ComponentChain.php" lineno="61"/>
  <stack where="Neos\Flow\Http\Component\ComponentChain->handle" level="2" type="file" filename="file:///var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Component_ComponentChain.php" lineno="61"/>
  <stack where="Neos\Flow\Http\RequestHandler->handleRequest" level="3" type="file" filename="file:///var/www/html/Packages/Framework/Neos.Flow/Classes/Http/RequestHandler.php" lineno="112"/>
  <stack where="Neos\Flow\Core\Bootstrap->run" level="4" type="file" filename="file:///var/www/html/Packages/Framework/Neos.Flow/Classes/Core/Bootstrap.php" lineno="109"/>
  <stack where="{main}" level="5" type="file" filename="file:///var/www/html/Web/index.php" lineno="27"/>
</response>

Umpa Lumpa can help you, he know the mapping
>>> /var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/GoCardTeam_GoCardApi_Http_v1_CORSInterceptionComponent.php
>>> /var/www/html/Packages/Application/GoCardTeam.GoCardApi/Classes/Http/v1/CORSInterceptionComponent.php

Umpa Lumpa can help you, he know the mapping
>>> /var/www/html/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Flow_Http_Component_ComponentChain.php
>>> /var/www/html/Packages/Framework/Neos.Flow/Classes/Http/Component/ComponentChain.php

Processed protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="stack_get" transaction_id="21">
  <stack where="GoCardTeam\GoCardApi\Http\v1\CORSInterceptionComponent->handle" level="0" type="file" filename="file:///var/www/html/Packages/Application/GoCardTeam.GoCardApi/Classes/Http/v1/CORSInterceptionComponent.php" lineno="25"/>
  <stack where="Neos\Flow\Http\Component\ComponentChain->handle" level="1" type="file" filename="file:///var/www/html/Packages/Framework/Neos.Flow/Classes/Http/Component/ComponentChain.php" lineno="61"/>
  <stack where="Neos\Flow\Http\Component\ComponentChain->handle" level="2" type="file" filename="file:///var/www/html/Packages/Framework/Neos.Flow/Classes/Http/Component/ComponentChain.php" lineno="61"/>
  <stack where="Neos\Flow\Http\RequestHandler->handleRequest" level="3" type="file" filename="file:///var/www/html/Packages/Framework/Neos.Flow/Classes/Http/RequestHandler.php" lineno="112"/>
  <stack where="Neos\Flow\Core\Bootstrap->run" level="4" type="file" filename="file:///var/www/html/Packages/Framework/Neos.Flow/Classes/Core/Bootstrap.php" lineno="109"/>
  <stack where="{main}" level="5" type="file" filename="file:///var/www/html/Web/index.php" lineno="27"/>
</response>


IDE >>> Debugger
================
Raw protocol:
context_names -i 22

Processed protocol:
context_names -i 22


Debugger >>> IDE
================
Raw protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="context_names" transaction_id="22">
  <context name="Locals" id="0"/>
  <context name="Superglobals" id="1"/>
  <context name="User defined constants" id="2"/>
</response>

Processed protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="context_names" transaction_id="22">
  <context name="Locals" id="0"/>
  <context name="Superglobals" id="1"/>
  <context name="User defined constants" id="2"/>
</response>


IDE >>> Debugger
================
Raw protocol:
context_get -i 23 -d 0 -c 0

Processed protocol:
context_get -i 23 -d 0 -c 0


Debugger >>> IDE
================
Raw protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="context_get" transaction_id="23" context="0">
  <property name="$componentContext" fullname="$componentContext" type="object" classname="Neos\Flow\Http\Component\ComponentContext" children="1" numchildren="3" page="0" pagesize="100">
    <property name="httpRequest" fullname="$componentContext->httpRequest" facet="protected" type="object" classname="Neos\Flow\Http\Request" children="1" numchildren="11"/>
    <property name="httpResponse" fullname="$componentContext->httpResponse" facet="protected" type="object" classname="Neos\Flow\Http\Response" children="1" numchildren="8"/>
    <property name="parameters" fullname="$componentContext->parameters" facet="protected" type="array" children="0" numchildren="0"/>
  </property>
  <property name="$response" fullname="$response" type="uninitialized"/>
  <property name="$this" fullname="$this" type="object" classname="GoCardTeam\GoCardApi\Http\v1\CORSInterceptionComponent" children="0" numchildren="0" page="0" pagesize="100"/>
</response>

Processed protocol:
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="context_get" transaction_id="23" context="0">
  <property name="$componentContext" fullname="$componentContext" type="object" classname="Neos\Flow\Http\Component\ComponentContext" children="1" numchildren="3" page="0" pagesize="100">
    <property name="httpRequest" fullname="$componentContext->httpRequest" facet="protected" type="object" classname="Neos\Flow\Http\Request" children="1" numchildren="11"/>
    <property name="httpResponse" fullname="$componentContext->httpResponse" facet="protected" type="object" classname="Neos\Flow\Http\Response" children="1" numchildren="8"/>
    <property name="parameters" fullname="$componentContext->parameters" facet="protected" type="array" children="0" numchildren="0"/>
  </property>
  <property name="$response" fullname="$response" type="uninitialized"/>
  <property name="$this" fullname="$this" type="object" classname="GoCardTeam\GoCardApi\Http\v1\CORSInterceptionComponent" children="0" numchildren="0" page="0" pagesize="100"/>
</response>


IDE >>> Debugger
================
Raw protocol:
context_get -i 24 -d 0 -c 1

Processed protocol:
context_get -i 24 -d 0 -c 1


Debugger >>> IDE
================
xml.Decoder.Token() - XML syntax error on line 1: unexpected EOF
...
```

**Comment by dfeyer** [comment]
This should be fixed in the recent release 1.0.2
